### PR TITLE
Resurrect the Interfaces ZMI tab

### LINKED ISSUE #450: "Interfaces" tab gone?
Is there a reason why the Interfaces tab was removed from the Zope 4  ZMI.

### PR DESCRIPTION
**Pull request by dataflake**
Fixes #450

### CODE REVIEW

File: src/Products/Five/utilities/browser/marker.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+        self.adapted = IMarkerInterfaces(context)
+        self.context_url = self.context.absolute_url()
+
+    def __call__(self, SAVE=None, add=(), remove=()):
+        if SAVE:
+            self.update(add, remove)
+            self.request.response.redirect(self.request.ACTUAL_URL)
```
Comment (icemac) [inline src/Products/Five/utilities/browser/marker.py]: Setting a `manage_tabs_message` here would be nice to hint the user that his save request was successful.

File: src/Products/Five/utilities/browser/marker.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            return ''
+        return self.index()
+
+    def _getLinkToInterfaceDetailsView(self, interfaceName):
+        return (self.context_url + (
+            '/views-details.html?iface=%s&type=zope.publisher.'
+            'interfaces.browser.IBrowserRequest' % interfaceName))
```
Comment (icemac) [inline src/Products/Five/utilities/browser/marker.py]: Where is this link used? I was not able to find it in the generated HTML code on `manage_interfaces` and there seems to be no view named `views-details.html` in Zope.

**Comment by dataflake** [comment]
I simply copied the code from the moment it was removed by Hanno. I did not go over it with a fine tooth comb, I just made sure the ZMI part works as intended.

There was a test module, see https://github.com/zopefoundation/Zope/commit/c4c1b4acf51b9d6f2b699801373d750b50e74673. This test module is a doctest. I spent an hour trying to make it work but couldn't, due to the fact that doctests are utterly impossible to debug in a sane way. I'm not going to spend any more time on that module.

I'm not invested in this PR. This was just in reaction to complaints in #450 and the fact that several people replied wanting it back.

File: src/Products/Five/utilities/browser/marker.py (hunk 1 above)
Comment (dataflake) [reply src/Products/Five/utilities/browser/marker.py]: You're right, I added that now.

File: src/Products/Five/utilities/browser/marker.py (hunk 2 above)
Comment (dataflake) [reply src/Products/Five/utilities/browser/marker.py]: I removed that method now.

**Comment by dataflake** [comment]
@icemac After converting the tests from &%$^ doctests I had a chance at debugging and fixing them, they are back and passing now. Take another look when you have a minute.

**Pull request merged**